<launch>
    <!-- Set arguments. Change values in this section to control various parameters of execution. -->
    <!-- There are some parameters that can be changed in other launch files but you should generally leave them as they are -->
    <!-- ********** START OF SECTION ********** -->
    <arg name="map_name" default="borongaj"/>       <!-- Name of the map used. -->
    <arg name="ctrl_loop_freq" default="10"/>       <!-- Frequency used by Reynolds rules. -->
    <arg name="data_stream_freq" default="100"/>    <!-- Position streaming frequency, used by Kalman filter. -->
    <arg name="debug_boids" default="false"/>       <!-- Enable debugging for Reynolds controller node. -->
    <arg name="debug_kalman" default="false"/>      <!-- Enable debugging for Kalman filter node. -->
    <arg name="wait_time" default="0"/>             <!-- During first X seconds of execution, no velocity commands are sent to robots. -->
    <arg name="start_time" default="2"/>            <!-- During first X seconds after "wait_time", inital velocity commands are sent to robots. -->
    <!-- *********** END OF SECTION *********** -->

    <arg name="map_yaml" default="$(find sphero_formation)/resources/maps/$(arg map_name).yaml"/>

    <!-- Set arguments as ros parameter so all nodes can access them. -->
    <param name="ctrl_loop_freq" type="int" value="$(arg ctrl_loop_freq)"/>
    <param name="data_stream_freq" type="int" value="$(arg data_stream_freq)"/>
    <param name="debug_boids" type="boolean" value="$(arg debug_boids)"/>
    <param name="debug_kalman" type="boolean" value="$(arg debug_kalman)"/>
    <param name="wait_time" type="double" value="$(arg wait_time)"/>
    <param name="start_time" type="double" value="$(arg start_time)"/>
    <param name="run_type" type="string" value="real"/>
    <param name="data_associated" type="string" value="false"/>

    <!-- Start map server. -->
    <node pkg="map_server" type="map_server" args="$(arg map_yaml)" name="map_server"/>
    <!-- Start rqt GUI and dynamic reconfigure node. -->
    <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_dynamic_reconfigure"/>
    <node pkg="sphero_formation" type="dynamic_reconfigure_node.py" name="dyn_reconf" output="screen"/>
    <!-- Start rviz. -->
    <param name="robot_description" textfile="$(find sphero_formation)/resources/simple_ball.urdf" />
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find sphero_formation)/launch/real/sphero_real.rviz"/>
</launch>
